dialects/hir/src/ops/binary.rs

```diff
@@ -573,6 +573,22 @@ pub struct Neq {
 
 infer_return_ty_for_binary_op!(Neq as Type::I1);
 
+/// Inequality comparison with immediate
+#[operation(
+    dialect = HirDialect,
+    implements(InferTypeOpInterface)
+)]
+pub struct NeqImm {
```

Comment:
We shouldn't require these types of operations anymore (i.e. variants that accept an explicit immediate argument). The operand folding infrastructure can match on operations with constant operands and fold them where appropriate, and in all other circumstances will tell us whether a given operand is immediate or not.

The fewer op variants we have, the better, but this in particular is one of the ways the new IR is nicer than the old one. To be clear, you can still have builder methods like `neq_imm`, just have it emit both a constant op and a `neq`

Reply:
Got it. I'll remove them ASAP. 
I was suspected this when I saw `*Imm` ops missing, but then I found `ShlImm`. :)